### ISSUE TITLE
evdi: [D] evdi_detect:69 (dev=2) Painter is connected / evdi: [D] evdi_painter_get_edid_copy:192 (dev=2) 00 ff ff

### ISSUE THREAD

**Issue by nekromoff** [DisplayLink/evdi#66]
repeated messages, no output to external monitor, even when xrandr can see it:

[  355.136811] evdi: [D] evdi_detect:69 (dev=2) Painter is connected
[  355.136815] evdi: [D] evdi_painter_get_edid_copy:192 (dev=2) 00 ff ff
[  355.613540] evdi: [D] evdi_detect:69 (dev=2) Painter is connected
[  355.613557] evdi: [D] evdi_painter_get_edid_copy:192 (dev=2) 00 ff ff
[  355.636036] evdi: [D] evdi_detect:69 (dev=2) Painter is connected
[  355.636043] evdi: [D] evdi_painter_get_edid_copy:192 (dev=2) 00 ff ff
[  355.643900] evdi: [D] evdi_detect:69 (dev=2) Painter is connected
[  355.643905] evdi: [D] evdi_painter_get_edid_copy:192 (dev=2) 00 ff ff
[  355.649452] evdi: [D] evdi_detect:69 (dev=2) Painter is connected
[  355.649456] evdi: [D] evdi_painter_get_edid_copy:192 (dev=2) 00 ff ff

**Comment by taitai222** [comment]
I have the same problem.
My kernel version :                    4.5.2
PC :                                           Microsoft Surface Pro 4

**Comment by displaylink-mlukaszek** [comment]
Is this also happening with the very latest evdi which now ships in 1.3 Ubuntu driver? Are you using open-source drivers for built-in screen? (e.g. nouveau for nvidia?)

**Comment by ethermeister** [comment]
Same issue here, slightly different behaviour. Screen gets painted once but no more. Cursor is visible on the secondary display.

From journalctl:

`	Mar 03 10:10:57 nooma kernel: evdi: [D] evdi_detect:75 (dev=1) Painter is connected
	Mar 03 10:10:57 nooma kernel: evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
	Mar 03 10:10:59 nooma kernel: evdi: [D] evdi_crtc_cursor_set:238 evdi_crtc_cursor_set unlock
	Mar 03 10:11:01 nooma kernel: evdi: [D] evdi_crtc_cursor_set:238 evdi_crtc_cursor_set unlock`

This is the same when using 1.3.52 or 1.2.64 rpm. Running on kernel 4.8.15-300.fc25.x86_64 (evdi fails to load on latest 4.9 kernel). Xrandr detects the display correctly. SELinux disabled on my tests.

Hardware is Dell XPS 13 9360 (Kaby Lake i7-7500U), with stock Intel HD graphics.

**Comment by diegoroccia** [comment]
Hello everyone.
after the last upgrade, I'm experiencing the exact same problem @ethermeister does.
secondary ( external ) screen is on, I can move the mouse but no desktop there. 
Dell XPS 13 9360. Ubuntu (17.04 RC)  Gnome (3.24) . did you find a solution?

**Comment by emilianobonassi** [comment]
me too!

**Comment by ment0s** [comment]
Give it a go with this ...  

/etc/X11/xorg.conf.d/20-intel.conf 
Section "Device"
Identifier "intel"
Driver "modesetting"
Option "kmsdev" "/dev/dri/card0"
Option "PageFlip" "off"
#Option "SWCursor" "on"
Option "ShadowFB" "true"
EndSection

Section "Device"
Identifier "USB3"
BusID "USB"
Driver "modesetting"
Option "kmsdev" "/dev/dri/card1"
Option "PageFlip" "off"
#Option "SWCursor" "on"
Option "ShadowFB" "true"
EndSection

**Comment by emilianobonassi** [comment]
Thanks @ment0s  but it doesn't work for me.

**Comment by dawikur** [comment]
Have you tried that: http://support.displaylink.com/knowledgebase/articles/1181623 ?

**Comment by emilianobonassi** [comment]
Thanks @dawikur , I'm using this https://github.com/DisplayLink/evdi/issues/85#issuecomment-289511245 which seems to be similar to use `Driver intel` as indicated in your link. Do you think it's not?

**Comment by dawikur** [comment]
@emilianobonassi  It is quite similar ;) The essential part of this version is the 'Driver "Intel', have you tried without all other settings?

**Comment by emilianobonassi** [comment]
@dawikur , I've just tried following your link, anyway it still doesn't work.
I continue to receive the following messages:
```
[   10.290882] evdi: [W] evdi_painter_disconnect:462 (dev=0) An unknown connection to ffff9230e4f0d400 tries to close us
[   10.290883] evdi: [W] evdi_painter_disconnect:463  - ignoring
[   10.308977] evdi: [W] evdi_painter_disconnect:462 (dev=0) An unknown connection to ffff9230e8678000 tries to close us
[   10.308979] evdi: [W] evdi_painter_disconnect:463  - ignoring
```

**Comment by dawikur** [comment]
What is your system? What session are you using?

**Comment by emilianobonassi** [comment]
Ubuntu 17.04 stock on Dell XPS 9360
Kernel
```
Linux NB-EBonassi 4.10.0-26-generic #30-Ubuntu SMP Tue Jun 27 09:30:12 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
Video
```
lightdm 1.22.0-0ubuntu2.1
xorg 1:7.7+16ubuntu3
gnome-session 3.24.0-0ubuntu1
```

Hope it helps. Tell me if you need some specific log.

**Comment by emilianobonassi** [comment]
Just solved installing latest kernel version (4.12) from the official PPA (but it's not officially supported) http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.12/

**Comment by dawikur** [comment]
Great news 👍 Thanks for letting us know.
